addons/l10n_fr_account/models/account_move.py

```diff
@@ -19,3 +19,18 @@ def _get_view(self, view_id=None, view_type='form', **options):
     def _compute_l10n_fr_is_company_french(self):
         for record in self:
             record.l10n_fr_is_company_french = record.country_code in record.company_id._get_france_country_codes()
+
+    @api.depends('country_code', 'move_type')
+    def _compute_show_delivery_date(self):
+        # EXTENDS 'account'
+        super()._compute_show_delivery_date()
+        for move in self:
+            if move.country_code == 'FR':
+                move.show_delivery_date = move.is_sale_document()
+
+    def _post(self, soft=True):
+        res = super()._post(soft)
+        for move in self:
+            if move.country_code == 'FR' and move.is_sale_document() and not move.delivery_date:
```

Comment:
you can use the show_delivery_date delivery date instead of forcing this here, and we could argue that you must use the is_company_french since some country are part of France like monaco or some island :smile: